src/features/todos.ts

```diff
@@ -1,8 +1,41 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
+interface TodosState {
+  items: Todo[];
+  status: 'idle' | 'loading' | 'succeeded' | 'failed';
+}
+
+const initialState: TodosState = {
+  items: [],
+  status: 'idle',
+};
+
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    setTodos(state, action) {
+      // eslint-disable-next-line no-param-reassign
+      state.items = action.payload;
+      // eslint-disable-next-line no-param-reassign
+    },
+    setLoading(state) {
+      // eslint-disable-next-line no-param-reassign
+      state.status = 'loading';
+    },
+    setEndLoading(state) {
+      // eslint-disable-next-line no-param-reassign
+      state.status = 'idle';
+    },
+    setError(state) {
+      // eslint-disable-next-line no-param-reassign
+      state.status = 'failed';
+    },
```

Comment:
Similar to the `setLoading` reducer, the `setError` reducer seems to be responsible for setting the error state. It would be more appropriate to handle error states within async thunks, where you can catch any errors that occur during the async operation and set the error state accordingly.